server/gitrest/packages/gitrest-base/src/routes/summaries.ts

```diff
@@ -140,10 +140,10 @@ async function createSummary(
     Lumberjack.info("Creating summary", lumberjackProperties);
 
     const { isNew, writeSummaryResponse } = await wholeSummaryManager.writeSummary(payload, isInitialSummary);
+    lumberjackProperties["newDocument"] = isNew;
 
-    // Waiting to pre-compute and persist latest summary would slow down document creation,
-    // so skip this step if it is a new document.
-    if (!isNew && isContainerSummary(payload)) {
+    // Waiting to pre-compute and persist latest summary would slow down document creation.
+    if (isContainerSummary(payload)) {
```

Comment:
Hey @sharmakhushboo , I didn't get a chance to talk about this when we had the daily scrum earlier, but I remember @znewton and I thought about avoiding to persist the initial summary (uploaded during document creation) because that significantly slowed the create document API. Perhaps we can have a chat to discuss about this?

Reply:
Hey! Yes, I spoke to Zach and Yang about this.

The reason for not persisting initial Summary earlier was that we thought it was unnecessary to store initial summary. But given the scenario i explained during the sync, persisting the initial summary will improve the getSummary latency. Else the joinOp errors is leading to connectDoc latency. 
yes, it can slow down the createDoc api, for which we could make the persisting to storage a fire-n-forget later if need be.
There needs to be a tradeoff b/w create and connectDoc latency.

Reply:
I'll be doing a load test on this change to get numbers for connectFlow. That's why this is in test branch.

Reply:
Makes total sense! I was bringing it up so we can keep this in mind.

I think we might need to consider the volume of create document requests vs. the volume of connect document requests. Fire and forget was something we considered, but unfortunately, the work was not trivial. Can you include me in the conversation with Yang and Zach? 😄  so I can learn more about the scenario. Thanks in advance!

Reply:
Sure. I spoke to them separately. But as I discussed with Zach, the fire-n-forget change requires a bit more discussion, for which I plan on creating a thread. Will do it later, after my testing.